src/Toml/Bi/Code.hs

```diff
@@ -88,7 +89,12 @@ type TomlCodec a = BiCodec Env St a
 decode :: TomlCodec a -> Text -> Either DecodeException a
 decode codec text = do
     toml <- first ParseError (parse text)
-    runReader (runExceptT $ codecRead codec) toml
+    decodeToml codec toml
+
+-- | Convert toml into user data type.
+decodeToml :: TomlCodec a -> TOML -> Either DecodeException a
+decodeToml codec =
```

Comment:
`decodeToml` name looks like the obvious choice but I don't think that it's the right. If the module is imported qualified the this will be `Toml.decodeToml` which is kinda awkward... Also, `Toml` is kinda generic suffix, the existing `decode` function can be named `decodeToml` as well. The point here is that `Toml` suffix doesn't introduce hints on how to use the function.

One example of better name: `decodeTOML` -- same as `TOML` data type. But this is still bad.

I think perfect name is `runTomlCodec` because it kinda fits the naming scheme for monadic functions.

If you will implement the dual of `decodeToml`, like `pretty` but function that returns `TOML` instead of `Text`, you could name this function as `execTomlCodec`: like `execState` that returns end state.

Btw, adding `execTomlCodec` is also a good idea :+1:

Reply:
Thank you for explain!